packager/mux/mux_test.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mux
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	pkgt "github.com/ampproject/amppackager/packager/testing"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// Propagate hardcoded values into template test url.
+func Expand(templateURL string) string {
+	templateURL = strings.Replace(templateURL, "$HOST",
+		"http://www.publisher_amp_server.com", 1)
+	templateURL = strings.Replace(templateURL, "$FETCH",
+		"http://www.publisher_main_server.com/some_page", 1)
+	templateURL = strings.Replace(templateURL, "$SIGN",
+		"https://www.publisher_main_server.com/some_page", 1)
+	templateURL = strings.Replace(templateURL, "$CERT", pkgt.CertName, 1)
+	return templateURL
+}
+
+// Dedicated type for annotated params - for tests readability.
+type params map[string]string
+
+// Mock for underlying http handlers - signer, cert etc.
+type MockedHandler struct {
+	mock.Mock
+}
+
+// Mock ServeHTTP: record annotated params, don't forward the call.
+func (m *MockedHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	// Convert annotated parameters to dedicated "params" type for tests readability.
+	m.Called(params(Params(req)))
+}
+
+func TestServeHTTPSuccess(t *testing.T) {
+	tests := []struct {
+		expectedHandlerName  string
+		testScenario         string
+		url                  string
+		expectedParsedParams params
+	}{
+		{"signer",
+			" empty",
```

Comment:
Something odd about indentation here and below.  Try gofmt or goimports?

Reply:
Done. Also switched to field:value initializations for readability.